Cloud/Docker/DuktapeBuilder/Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM swift:latest
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV SHELL=bash
+
+RUN apt-get -y update && apt-get -y upgrade
+RUN apt-get install -y git make clang bc python-pip
+
+# Dependency of the duktape build process
+RUN pip install pyyaml
+
+RUN useradd -m builder
+
+RUN git clone https://github.com/svaarala/duktape.git /home/builder/duktape
+WORKDIR /home/builder/duktape
+
+# Docker will attempt to cache the output of every step. That's fine (and useful to speed things up, e.g. by avoiding
+# the need to download the entire source repository again every time!). However, whenever the following ARG is changed
+# (i.e. we are building a new version of the engine), a cache miss occurs (because the build context changed) and all
+# steps from here on are rerun. That, however, means we might be operating on an old checkout of the source code from
+# the cache, and so we update it again before checking out the requested revision.
+RUN git pull && git checkout $rev
```

Comment:
Aren't you missing the `ARG` here now? I think you can just copy lines 20 through 27 from https://github.com/googleprojectzero/fuzzilli/blob/5aa6f4ef3724f197a0fc55ffc9b8caf8c809ad56/Cloud/Docker/JSCBuilder/Dockerfile#L20 that should work

Reply:
Ah hmm thinking about it I guess the caching logic as described in that comment is broken if $rev is master (as it is in the build.sh for this), since then docker won't see a change and just completely use the cached image without updating to the latest source code. Anyway, I think you can just copy+paste those lines from the JSCBuilder, then I'll merge it and then I'll think of some way to fix the caching problem since it'll soon affect v8 as well (v8 can now also be built from master for fuzzing) :)

Reply:
How about something like this in build.sh? It's in the commit I just pushed.
```
REV=$(git ls-remote https://github.com/svaarala/duktape.git | grep refs/heads/master | awk '{print $1;}')
```
That'll always get the most recent commit on master, without having to separately download the repo outside the dockerfile.

I'm still pretty new at Docker, but I think that gets the caching behavior we're looking for.

Also, both `HEAD` and `refs/heads/master` have the same commit hash, at least in Duktape. I think the difference would matter if the main branch is no longer named `master`, but should be fine with the repos we're working with?

Reply:
I had to update the dockerfile to properly consume the rev argument. Once we figure out what exactly correct looks like here, I can squash the commits before the actual merge.